core/src/main/scala/kafka/server/ReplicaAlterLogDirsThread.scala

```diff
@@ -76,13 +83,69 @@ class ReplicaAlterLogDirsThread(name: String,
     futureLog.updateHighWatermark(partitionData.highWatermark)
     futureLog.maybeIncrementLogStartOffset(partitionData.logStartOffset, LogStartOffsetIncrementReason.LeaderOffsetIncremented)
 
-    if (partition.maybeReplaceCurrentWithFutureReplica())
-      removePartitions(Set(topicPartition))
+    directoryEventHandler match {
+      case DirectoryEventHandler.NOOP =>
+        if (partition.maybeReplaceCurrentWithFutureReplica())
+          removePartitions(Set(topicPartition))
+      case _ =>
+        maybePromoteFutureReplica(topicPartition, partition)
+    }
 
     quota.record(records.sizeInBytes)
     logAppendInfo
   }
 
+  override def removePartitions(topicPartitions: Set[TopicPartition]): Map[TopicPartition, PartitionFetchState] = {
+    // Schedule assignment request to revert any queued request before cancelling
+    for {
+      topicPartition <- topicPartitions
+      partitionState <- partitionAssignmentRequestState(topicPartition)
+      if partitionState == QUEUED
+      partition = replicaMgr.getPartitionOrException(topicPartition)
+      topicId <- partition.topicId
+      directoryId <- partition.logDirectoryId()
+      topicIdPartition = new TopicIdPartition(topicId, topicPartition.partition())
+    } yield directoryEventHandler.handleAssignment(topicIdPartition, directoryId)
```

Comment:
Seems odd to use `yield` when `directoryEventHandler.handleAssignment()` is a void method.  Is there a reason to use `yield`?

Reply:
No reason that I can call. I removed the `yield`